### ISSUE TITLE
Plugin gets in the way when introducing intermediate effects

### ISSUE THREAD

**Issue by isovector** [polysemy-research/polysemy#209]
```haskell
data View v m a where
  View :: View v m v

makeSem ''View


viewToState :: forall v s r a. Member (State s) r => (s -> Sem r v) -> Sem (View v ': r) a -> Sem r a
viewToState f m = do
  s0 <- get
  v0 <- f s0
  evalState v0
    $ intercept @(State s) @(State v ': r) undefined
    $ reinterpret @(View v) @(State v) undefined
    $ m
```

```
/home/sandy/prj/dynahaskell/src/Sem/View.hs:20:7: error:
    • Could not deduce: s ~ v arising from a use of ‘intercept’
      from the context: Member (State s) r
        bound by the type signature for:
                   viewToState :: forall v s (r :: [(* -> *) -> * -> *]) a.
                                  Member (State s) r =>
                                  (s -> Sem r v) -> Sem (View v : r) a -> Sem r a
        at /home/sandy/prj/dynahaskell/src/Sem/View.hs:15:1-101
```

:(

**Comment by isovector** [comment]
Maybe we could solve this with an annotation on `viewToState` that turns off the plugin locally.

**Comment by isovector** [comment]
Not actually solved by #245. that's embarassing!

**Comment by isovector** [comment]
Actually it _is_ solved by #245!
